numpy/core/src/multiarray/dtypemeta.c

```diff
@@ -698,6 +735,39 @@ object_common_dtype(
     return cls;
 }
 
+static int
+fill_zero_object_strided_loop(
+        void *NPY_UNUSED(traverse_context), PyArray_Descr *NPY_UNUSED(descr),
+        char *data, npy_intp size, npy_intp stride,
+        NpyAuxData *NPY_UNUSED(auxdata))
+{
+    PyObject *zero = PyLong_FromLong(0);
+    PyObject **optr = (PyObject **)data;
+    while (size--) {
+        Py_INCREF(zero);
+        *optr++ = zero;
+    }
+    Py_DECREF(zero);
+    if (PyErr_Occurred()) {
+        return -1;
+    }
```

Comment:
```suggestion
```
An error cannot occur and if it did we should have checked explicitly for it.  The only thing that may have a check is `zero = `.  (cannot fail on CPython, but not sure about PyPy)

Reply:
OK good point, FWIW this was taken from the old implementation of `_zerofill`, it probably wasn't necessary there either.